ssg/build_remediations.py

```diff
@@ -146,6 +151,81 @@ def get_populate_replacement(remediation_type, text):
     sys.exit(1)
 
 
+def from_file(file_path, env_yaml):
+    mod_file = []
+    config = defaultdict(lambda: None)
+
+    fix_file_lines = jinja_process_file(file_path, env_yaml).splitlines()
+
+    # Assignment automatically escapes shell characters for XML
+    for line in fix_file_lines:
+        line += "\n"
+        if line.startswith('#'):
+            try:
+                (key, value) = line.strip('#').split('=')
+                if key.strip() in ['complexity', 'disruption',
+                                   'platform', 'reboot',
+                                   'strategy']:
+                    config[key.strip()] = value.strip()
+                else:
+                    if not line.startswith(FILE_GENERATED):
+                        mod_file.append(line)
+            except ValueError:
+                if not line.startswith(FILE_GENERATED):
+                    mod_file.append(line)
+        else:
+            mod_file.append(line)
+
+    return mod_file, config
+
+
+def process_fix(fixes, args, env_yaml, product, file_path, fix_name):
+    if not is_supported_filename(args.remediation_type, file_path):
+        return
+
+    fix_contents, config = from_file(file_path, env_yaml)
+
+    if is_applicable_for_product(config['platform'], product):
+        fixes[fix_name] = (fix_contents, config)
+    elif not config['platform']:
+        raise RuntimeError(
```

Comment:
This exception could have been raised by some higher-level wrapper of `from_file`.

Reply:
I think this is the higher-level wrapper of `from_file`? I can't move it to `from_file` because I use `from_file` in the `rule_dirs_json` to parse the config element. And creating a wrapper for `from_file` just to check if `platform` exists in the config for use in this function doesn't make sense to me.

I moved this up to before the if statement that sets fixes so hopefully this is better.